### ISSUE TITLE
Commend Count Not Updating After Receiving New Commends in Matches

### ISSUE THREAD

**Issue by layer07** [ValveSoftware/Dota2-Gameplay#11278]
#### Description:

I have noticed that my commend count is not updating as expected after receiving new commends from other players. Despite actively asking for and receiving commends in post-match screens, the total commend count remains unchanged.

#### Steps to Reproduce:

1. Complete a match.
2. Ask for commends from teammates or opponents in the post-match screen.
3. Receive commends from other players.
4. Check the total commend count on the profile.

#### Expected Behavior:

The total commend count should increment based on the number of new commends received.

#### Actual Behavior:

The total commend count remains unchanged, even after receiving multiple new commends.

#### Additional Information:

- This issue has been persisting for approximately one week.
- I have played over 80 matches during this period and have received a significant number of commends.
- My current commend count is stuck at 8560.

I hope this issue can be resolved soon. Thank you for your attention.

**Comment by karadoulis** [comment]
Any updates on this one? They really don't care? 
Support said that they don't care and to open a ticket here on GitHub.
Since there are no money in the middle I don't think they gonna fix it.

**Comment by UzLA** [comment]
the commends counter stuck for several day (more like a week+) and I missed dozens of commends already because of this!

**Comment by karadoulis** [comment]
Ok today another update took place with a new look in the profile of the player but NO FIX FOR COMMENDS.
This is really a joke now because nobody cares about this issue.
Really poor support from valve.
But hey the international is now live and you will focus on how to take our money, so who cares right?

**Comment by ligaapl** [comment]
Yes. I have 2172 commends a month ago and commends still not counted after profile update! ID Steam 113184848

**Comment by kutaykartoz** [comment]
Can this be fixed already??? and fix it retroactively if possible..
This issue is here for more than a month. Valve even implemented new custom profiles and new behavior system, but didn't care about fixing one of its essential ingredient.

**Comment by commentator24** [comment]
I might be wrong but readme note said they prioritise sort issues by thumbs up reaction, so they might sort like this
https://github.com/ValveSoftware/Dota2-Gameplay/issues?q=is%3Aopen+is%3Aissue+sort%3Areactions-%2B1-desc
![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/109324155/b602b4c6-9a87-4865-88dc-690fafff3337)

is there any reason why you guys not upvoting the issue that you feel are important? its kinda funny to see you guys don't care about upvoting on original post, while they don't care about fixing the ones without upvotes. 11 similar threads yet only 2 upvotes 😆 
![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/109324155/fc4ec6d3-7930-4698-a362-7829cf018de7)
i think they should leave this broken to ensure bad behaviour players takes a longer time to get back to 12000 hehehe 😈

**Comment by trevordell** [comment]
I noticed this right away because my continued existence relies on seeing the commend counter go up every day.

With the new patch, I figured that it was in all prep for the new UI and mini-profiles, and that the _old_ way just wasn't pulling the stats properly. But nope, my profile and mini-profile displayed commends are still what they were 6 weeks ago.

**Comment by MinGthe13th** [comment]
Same problem. Commend count stucked at 4939. Although sometimes the loading screen shows I got like 124 commends in 15 games (which is obviously another bug) but the profile amount just didn't accumulate.

**Comment by layer07** [comment]
Still no word on this?

**Comment by rishab-bandekar** [comment]
What is the update on this?

**Comment by karadoulis** [comment]
I think the update is that Valve does not care because it hasn't over 100 reactions.
I contact support and clearly they don't care since this bug doesn't affect their money income.
Really poor support from Valve. Good to know that they only care is money.

**Comment by commentator24** [comment]
maybe they changed it to percentage value? #12658 e.g. max 100?

**Comment by trevordell** [comment]
> maybe they changed it to percentage value? #12658 e.g. max 100?

His stopped right around when mine stopped. (And most likely a whole bunch of others.)

I also have done over 180 games since it stopped. (Which would work out to a number much greater. Usually 1-4 per win, 0 is very rare unless the coordinator is down. And get the odd one in losses too.)

**Comment by zeekcox** [comment]
commends still not working

**Comment by WellNonato** [comment]
Hello, 

The recommendations number widget is not updating. I have received many commendations and none was imputed.

This party ID is just an example, maybe there are months it is happening and I did not get it. Other player possibly have the same problem.

Thank you very much.

Example Match ID (and possibly Timestamp)
7495939612

![image](https://github.com/ValveSoftware/Dota2-Gameplay/assets/24615985/100a5e00-13f2-464e-b274-7021d27d41c9)

**Comment by RickyArthurs** [comment]
Issue still present for me has been like this for months. Support just recommends verifying game integrity.. obviously doesn't fix the issue

**Comment by vitorallves** [comment]
My account has this problem, it doesn't update the commends. Is there any news on this topic?

**Comment by WellNonato** [comment]
Hi @robinh-valve, 

Please, some update here?

Thanks

**Comment by vedavyasr** [comment]
same issue here from the last 5 months

**Comment by commentator24** [comment]
maybe the commend counters r burden to their database? imagine every 40 min millions of commend counter increase.

**Comment by RickyArthurs** [comment]
> maybe the commend counters r burden to their database? imagine every 40 min millions of commend counter increase.

Not really significant, mmr has to be updated after every game and behaviour scores etc. and this does not account for some accounts counter fully functional while others not

**Comment by layer07** [comment]
## Attention Valve Developers:
### We urgently need an IncreaseCommendFactory Enterprise Grade done by tomorrow!

```c++
// Valve Enterprise Database Reference
class ValveEnterpriseDatabase {
public:
    void saveToDatabase(int commendCount) {
        // Simulate saving to the most enterprise-grade database ever
        // because we know how to make things complicated!
    }
};

namespace EnterpriseCommendCounter {

    class Commend {
    public:
        virtual void increase() = 0;
        virtual int getCount() const = 0;
    };

    class ConcreteCommend : public Commend {
        int count;
    public:
        ConcreteCommend() : count(0) {}
        void increase() override { count++; }
        int getCount() const override { return count; }
    };

    class CommendStrategy {
    public:
        virtual void execute(Commend& commend) = 0;
    };

    class IncrementStrategy : public CommendStrategy {
    public:
        void execute(Commend& commend) override {
            commend.increase();
        }
    };

    class CommendContext {
        std::unique_ptr<CommendStrategy> strategy;
        std::unique_ptr<Commend> commend;
        ValveEnterpriseDatabase valveDatabase; // Valve's incredibly complex enterprise database
    public:
        CommendContext() : 
            strategy(std::make_unique<IncrementStrategy>()), 
            commend(std::make_unique<ConcreteCommend>()) {}

        void process() {
            strategy->execute(*commend);
            valveDatabase.saveToDatabase(commend->getCount()); // Saving to the Valve Enterprise Database
        }

        int getCommendCount() const {
            return commend->getCount();
        }
    };

    // Attention Valve Developers: We Need an IncreaseCommendFactory Here!
    // Because complexity knows no bounds in Valve's world!
    class IncreaseCommendFactory {
    public:
        static std::unique_ptr<CommendStrategy> createIncrementStrategy() {
            // Introducing the ultimate IncreaseCommendFactory!
            // We specialize in producing overengineered strategies.
            return std::make_unique<IncrementStrategy>();
        }
    };

    class CommendCountFacade {
        CommendContext context;
    public:
        void incrementCommendCount() {
            context.process();
            // No more boring comments, this is enterprise level code.
        }
    };

}

```

In the immortal words of Abraham Lincoln regarding C++, the programming language favored by Valve™®© for its remarkable ability to:

> C++ language supports OO programming by providing support for abstraction, encapsulation, inheritance, polymorphism, metamorphism, symbolism, neopaganism, satanism, communism, non-determinism and severe masochism.

I've created an excellent piece of EnterpriseLevelCode for IncreaseCommendCount with a Strategy, a CommendContext and it is ready to fullfil the complex but intricate demands of the enterprise.

Someone at Valve™®© should be responsible for shuffling papers around someone's desk and get it working soon.

Can someone at Valve™®© please help us?
![abssm](https://github.com/ValveSoftware/Dota2-Gameplay/assets/34219623/2c61643b-513d-41e2-adc1-98fa9e0c38f4)

**Comment by WellNonato** [comment]
Dear error partners,

Looks like they’re having large issues to distribute your attention and this is one more.

![IMG_0907](https://github.com/ValveSoftware/Dota2-Gameplay/assets/24615985/1c8d9240-0ad0-4cf8-96f3-d9bffef37426)

**Comment by DaltonCaughell** [comment]
Have been having this issue for over a year now. Played hundreds of matches, but commends are stuck no matter what at 956. No idea why, my behavior score is perfect.

**Comment by WellNonato** [comment]
Guys, good news, my commendation widget back to work normally. Check your too.

The problem is almost six months of commend was lost. This is sad, but is updating now.

Thank you Valve developers.
BR.

**Comment by RickyArthurs** [comment]
Still no confirmed fix, my widget and friends are still not updating. Still stuck for almost a year now

**Comment by layer07** [comment]
That is weird. Is it fixed or not?

**Comment by commentator24** [comment]
mine seems working. i remember it was 17k when it was broken
i'll screenshot again if its increasing
![image](https://github.com/user-attachments/assets/8b644e79-aa60-44b7-9710-66d253f0d691)

**Comment by DaltonCaughell** [comment]
Mine appears to be working as well, seems fixed

**Comment by layer07** [comment]
Help me with the Immortal account buyers issue plz.

https://github.com/ValveSoftware/Dota2-Gameplay/issues/22945

**Comment by commentator24** [comment]
yup fixed mine increasing. maybe just you / your server? what server you play
![image](https://github.com/user-attachments/assets/4b780e1f-8406-4cc1-9d42-4cb8027ccdcd)

**Comment by layer07** [comment]
Did it return again? 6 commends from yesterday match not counted.
